gremlin-go/driver/graphBinary.go

```diff
@@ -873,10 +877,51 @@ func enumReader(buffer *bytes.Buffer, _ *graphBinaryTypeSerializer) (interface{}
 	return string(valBytes), err
 }
 
+// Format: {key}{value}
+func bindingWriter(value interface{}, buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer) ([]byte, error) {
+	var v Binding
+	if reflect.TypeOf(value).Kind() == reflect.Ptr {
+		v = *(value.(*Binding))
+	} else {
+		v = value.(Binding)
+	}
+
+	// Not fully qualified.
+	_, err := typeSerializer.writeValue(v.Key, buffer, false)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = typeSerializer.write(v.Value, buffer)
+	if err != nil {
+		return nil, err
+	}
+	return buffer.Bytes(), nil
+}
+
+func bindingReader(buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer) (interface{}, error) {
+	b := new(Binding)
+
+	// Not fully qualified.
+	newKey, err := typeSerializer.readValue(buffer, byte(StringType), false)
+	if err != nil {
+		return nil, err
+	}
+	b.Key = newKey.(string)
+
+	newValue, err := typeSerializer.read(buffer)
+	if err != nil {
+		return nil, err
+	}
+	b.Value = newValue
+
+	return b, nil
+}
+
 // gets the type of the serializer based on the value
 func (serializer *graphBinaryTypeSerializer) getSerializerToWrite(val interface{}) (*graphBinaryTypeSerializer, error) {
 	switch val.(type) {
-	case bytecode, *GraphTraversal:
+	case *bytecode, bytecode, *GraphTraversal:
```

Comment:
Why is this required?

Reply:
In the `convertArgument` function, Traversal/GraphTraversals has `*bytecode` in its struct, so it returns the pointer. It was only the default case that returned a `bytecode`, and we actually was not hitting the `Traversal` case previously, because our traversals are returned as pointers(`*GraphTraversal`), so `bytecode` serialization was enough, but with bindings I need the case that returns the `*bytecode`.